### ISSUE TITLE
Guidance on updating to tfa AdamW from contribs AdamW

### ISSUE THREAD

**Issue by tsrobinson** [tensorflow/addons#495]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS
- TensorFlow version and how it was installed (source or binary):  2.0.0rc
- TensorFlow-Addons version and how it was installed (source or binary): 0.5
- Python version: 3.7.3
- Is GPU used? (yes/no): No

Hi, please tell me if this is the wrong place to post this issue. I'm in the process of updating my code to 2.0, and running into issues using the tfa AdamWOptimizer minimize function. 

Specifically, whereas `loss` used to take a Tensor, it now requires a callable. Could you please provide guidance/documentation on how to update from contrib.AdamWOptimizer to the tfa-version, with respect to this loss-function?

In tf 1.14.0 I had working code like the following:
```
self.joint_loss = tf.reduce_mean(input_tensor=tf.reduce_sum(input_tensor=cost_list))
# ...
tf.contrib.opt.AdamWOptimizer(lmbda, self.learn_rate)
self.train_step = optim.minimize(self.joint_loss)
```

But now using tfa, I get the error that 'Tensor' object is not callable:
```
self.joint_loss = tf.reduce_mean(input_tensor=tf.reduce_sum(input_tensor=cost_list))
# ...
optim = tfa.optimizers.AdamW(learning_rate=self.learn_rate, weight_decay=lmbda)
var_list = tf.compat.v1.trainable_variables()
self.train_step = optim.minimize(self.joint_loss, var_list  =  var_list)

```

**Comment by seanpmorgan** [comment]
Hi @tsrobinson! Without seeing the full code it's a little tough, but in general in TF2 I would recommend you run your code in eager execution or have Keras compile and fit the model.

As you said Keras optimizers need a callable loss to execute:
https://github.com/tensorflow/tensorflow/blob/r1.14/tensorflow/python/keras/optimizer_v2/optimizer_v2.py#L99
```
opt = tf.keras.optimizers.SGD(learning_rate=0.1)
  model = tf.keras.Sequential()
  model.add(tf.keras.layers.Dense(num_hidden, activation='relu'))
  model.add(tf.keras.layers.Dense(num_classes, activation='sigmoid')
  loss_fn = lambda: tf.keras.losses.mse(model(input), output)
  var_list_fn = lambda: model.trainable_weights
  for input, output in data:
    opt.minimize(loss_fn, var_list_fn)
```

This example nicely shows you what is happening behind the scene in `opt.minimize` 
https://www.tensorflow.org/beta/guide/eager
```
def train_step(images, labels):
  with tf.GradientTape() as tape:
    logits = mnist_model(images, training=True)
    
    loss_value = loss_object(labels, logits)
  loss_history.append(loss_value.numpy().mean())
  grads = tape.gradient(loss_value, mnist_model.trainable_variables)
  optimizer.apply_gradients(zip(grads, mnist_model.trainable_variables))
```

Or you can see how we utilize TFA optimizers using Keras compile in this example:
https://github.com/tensorflow/addons/blob/master/examples/optimizers_lazyadam.ipynb

**Comment by facaiy** [comment]
Hi, Tom, please check out https://www.tensorflow.org/beta/guide/eager#variables_and_optimizers and let us know if it helps :-)

**Comment by seanpmorgan** [comment]
Closing. Please feel free to re-open if this was not sufficient
